src/messaging/incoming.ts

```diff
@@ -27,9 +26,11 @@ export type NavigateEvent = {
 export type NavigateMessage = TestBoxMessage<typeof NAVIGATE, NavigateEvent>;
 
 export type LoginEvent = {
-  username?: string;
-  password?: string;
-  totpCode?: string;
+  email: string;
+  password: string | null;
```

Comment:
should we do :? string | null for each of these?

Reply:
hmmm, may be more clear